application/pages/default/ConsumePage.php

```diff
@@ -39,20 +39,35 @@ public function beforeView()
         usort($byFirstName, function($a, $b) {
             return strcmp(strtolower($a[User::FIELD_FIRST_NAME]), strtolower($b[User::FIELD_FIRST_NAME]));
         });
+        $byFirstName = array_filter($byFirstName, function($a) {
+            $timeStamp = strtotime($a['time']);
+            $cutOffDate = strtotime('-6 months');
+
+            return ($timeStamp > $cutOffDate);
+        });
+
         $byLastName = $allUsersByRole[Role::ROLE_USER];
         usort($byLastName, function($a, $b) {
             return strcmp(strtolower($a[User::FIELD_LAST_NAME]), strtolower($b[User::FIELD_LAST_NAME]));
         });
+        $byLastName = array_filter($byLastName, function($a) {
+            $timeStamp = strtotime($a['time']);
+            $cutOffDate = strtotime('-6 months');
+
+            return ($timeStamp > $cutOffDate);
+        });
```

Comment:
Code quality wise I would have factored out the filter function into a separate named function. Then both filters can use the same logic twice. 

DRY for only two instances of the code is often just an edge care. So I'm fine keeping it like this. More of a general remark.